06-building-decays.md

```diff
@@ -0,0 +1,142 @@
+> It also makes the output locations of the data written by the head of the selection chain available via the `outputLocations` method.
+
+To get our input particles we have two equivalent options:
+
+ - Use the `DataOnDemand` class with the `Phys/StdAllNoPIDsPions/Particles` and `Phys/StdAllLooseKaons/Particles` as inputs for pions and kaons, respectively.
+
+```python
+from PhysSelPython.Wrappers import DataOnDemand
+Pions = DataOnDemand('Phys/StdAllNoPIDsPions/Particles')
+Kaons = DataOnDemand('Phys/StdAllLooseKaons/Particles')
+```
+
+ - Load the equivalent particle creation algorithms from the `CommonParticles` package:
+
+```python
+from CommonParticles.StdAllNoPIDsPions import StdAllNoPIDsPions as Pions
+from CommonParticles.StdAllLooseKaons import StdAllLooseKaons as Kaons
+```
+
+The `CommonParticles` package, which you can find [here](https://svnweb.cern.ch/trac/lhcb/browser/Stripping/trunk/Phys/CommonParticles/python/CommonParticles), allows to access premade particles with reasonable selections and gives easy access to common particles used in selections.
+If you access the code for `StdAllNoPIDsPions`, you can see that in it the algorithm for creating the particles is instantiated and the map of the `DataOnDemand` service is updated via the `updateDoD` function.
+
+Once we have the input pions and kaons, we can combine them to build a D0 by means of the `CombineParticles` algorithm.
+This algorithm performs the combinatorics for us according to a given decay descriptor and puts the resulting particle in the TES, allowing also to apply some cuts on them:
+
+ - `DaughtersCuts` is a dictionary that maps each child particle we want to cut on with a selection LoKi particle functor (or combination of functors) which results in a `bool`. Optionally, one can specify also a `Preambulo` property that allows us to make imports, preprocess functors, etc (more on this in the [LoKi functors](https://lhcb.github.io/first-analysis-steps/06-loki-functors.html) lesson). For example:
+
+```python
+d0_daughters = {'pi-': '(PT > 750*MeV) & (P > 4000*MeV) & (MIPCHI2DV(PRIMARY) > 4)',
+                'K+': '(PT > 750*MeV) & (P > 4000*MeV) & (MIPCHI2DV(PRIMARY) > 4)'}
+```
+
+ - `CombinationCut` is a particle array LoKi functor (note the `A` prefix, see more [here](https://twiki.cern.ch/twiki/bin/view/LHCb/LoKiHybridFilters#Particle_Array_Functors)) that is given as input the array of particles in a single combination, in our case a kaon and a pion. This cut is performed the vertex fit, so typically can be used to save CPU cycles by performing some sanity cuts such as `AMAXDOCA` or `ADAMASS` (do you know what they do?) before the vertex fit is performed:
+ 
+```python
+d0_comb = "(AMAXDOCA('') < 0.2*mm) & (ADAMASS('D0') < 100*MeV)"
+```
+ 
+ - `MotherCut` is a selection LoKi particle functor that acts on the parent particle after the vertex fit, which allows to further filter on those variables that require the particle to be built:
```

Comment:
Briefly explain that parent particle == the particle made from the input particles according to the decay descriptor. Back in the day it took me a while to grasp that this was the particle that was just made from the inputs.